clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2119,17 +2085,16 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     return true;
   }
 
-  // FIXME: Refactor accessor handling. There is a discrepancy in how
-  // inherited accessors are handled.
-  bool handleSyclAccessorType(const CXXRecordDecl *, const CXXBaseSpecifier &BS,
-                              QualType FieldTy) final {
+  bool handleSyclSpecialType(const CXXRecordDecl *, const CXXBaseSpecifier &BS,
+                             QualType FieldTy) final {
     const auto *RecordDecl = FieldTy->getAsCXXRecordDecl();
-    assert(RecordDecl && "The accessor/sampler must be a RecordDecl");
+    assert(RecordDecl && "The type must be a RecordDecl");
     llvm::StringLiteral MethodName = KernelDecl->hasAttr<SYCLSimdAttr>()
                                          ? InitESIMDMethodName
                                          : InitMethodName;
-    CXXMethodDecl *InitMethod = getMethodByName(RecordDecl, MethodName);
-    assert(InitMethod && "The accessor/sampler must have the __init method");
+    CXXMethodDecl *InitMethod =
+        isCXXRecordWithInitOrFinalizeMember(RecordDecl, MethodName);
+    assert(InitMethod && "The type must have the __init method");
 
     // Get access mode of accessor.
     const auto *AccessorSpecializationDecl =
```

Comment:
If streams/samplers are used as base classes now, I think code will now crash since all special classes are now going through this handler. Prior to this change,  we just silently ignored it